core-rust/state-manager/src/jni/common_types.rs

```diff
@@ -0,0 +1,77 @@
+ *
+ * The Licensor does not make or purport to make and hereby excludes liability for all
+ * and any representation, warranty or undertaking in any form whatsoever, whether express
+ * or implied, to any entity or person, including any representation, warranty or
+ * undertaking, as to the functionality security use, value or other characteristics of
+ * any distributed ledger nor in respect the functioning or value of any tokens which may
+ * be created stored or transferred using the Work. The Licensor does not warrant that the
+ * Work or any use of the Work complies with any law or regulation in any territory where
+ * it may be implemented or used or that it will be appropriate for any specific purpose.
+ *
+ * Neither the licensor nor any current or former employees, officers, directors, partners,
+ * trustees, representatives, agents, advisors, contractors, or volunteers of the Licensor
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+use sbor::{Categorize, Decode, Encode};
+
+#[derive(Debug, PartialEq, Eq, Categorize, Encode, Decode)]
+pub struct JavaHashCode(Vec<u8>);
```

Comment:
Not sure why we need this on the Rust side? Why don't we map that directly to `Vec<u8>`? We use `HashCode` on the java side as an immutable wrapper for any byte arrays

Reply:
I actually did not introduce this type (it previously lived in `mempool.rs` [here](https://github.com/radixdlt/babylon-node/pull/339/files/18cc968b91db281017cd7480609090177cb7b416#diff-e5cb5c996dff03df099b84a17a7dbd51eeff543d873ddd5b58abd7f978ba8163L211), and I moved it to re-use it).

I thought it has to be `JavaHashCode` in order to be mapped to `HashCode` in Java :D but even if not technically needed, I think it's nice to have a well-named type here (not every `Vec<u8>` is conceptually a hash code), assuming it is cheap in runtime.